File: src/visualizations/Question 3 visualizations.py
# Databricks notebook source
import boto3
import pandas as pd

# COMMAND ----------

s3 = boto3.client('s3')

# COMMAND ----------

#import data
df = spark.read.load("s3n://group1-gr5069/processsed/preds_second_place.csv")
df = df.toPandas()

# COMMAND ----------

display(df)

# COMMAND ----------

df['pred_correct'] = (df['predictions']==df['driverRef']).astype(int)

# COMMAND ----------

display(df)

# COMMAND ----------

# MAGIC %md #### Save to RDS

# COMMAND ----------

#convert to spark dataframe
predictions_final = spark.createDataFrame(df)

# COMMAND ----------

predictions_final.write.format('jdbc').options(
      url='jdbc:mysql://sp-gr5069.ccqalx6jsr2n.us-east-1.rds.amazonaws.com/sp_test',
      driver='com.mysql.jdbc.Driver',
      dbtable='second_place_preds',
      user='admin',
      password='VgCrEPeYKWaaIZZhoYHt').mode('overwrite').save()

File: src/visualizations/Question 3 Visualizations.R
# Databricks notebook source
from pyspark.sql.types import DoubleType
from pyspark.ml.regression import LinearRegression
from pyspark.ml.feature import VectorAssembler
from pyspark.sql.functions import datediff, current_date, avg
from pyspark.sql.types import IntegerType
predDF = spark.read.csv('s3://group1-gr5069/processed/constructorMergeModels/part-00000-tid-932161557840838493-6910be16-f59b-495a-9846-6216092d48a8-90-1-c000.csv', header=True)
predDF.write.format('jdbc').options(
      url='jdbc:mysql://il-gr5069.ccqalx6jsr2n.us-east-1.rds.amazonaws.com/ian_test',
      driver='com.mysql.jdbc.Driver',
      dbtable='predDF',
      user='admin',
      password='v0lleyball').mode('overwrite').save()

# COMMAND ----------

